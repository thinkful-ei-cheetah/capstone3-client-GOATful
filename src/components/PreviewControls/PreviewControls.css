.preview-controls-container {
  width: 20vw;
  height:91vh;
  padding:10px;
  min-width:190px;
  /* height: 100vh; */
  border-right: 1px solid grey;
  margin: 0;
  overflow:scroll;
}

.preview-tiles {
  list-style-type: none;
  padding: 0;
  overflow-x: scroll;
}

.preview-title {
  text-align: center;
  margin: 0;
  padding: 1em;
  border-bottom: 1px solid grey;
  font-size: 1.3rem;
  margin-bottom:10px;
}

::-webkit-scrollbar { 
  /* display: none;  */
}

@media screen and (max-width: 630px) {
  .preview-controls-container {
    width: 100%;
    height:auto;
    border-right: none;
    border-bottom: 1px solid grey;
  }

  .preview-title {
    padding-bottom: 0;
    border: none;
    margin-bottom: 3px;
    z-index: 19;
  }
  
  
  .preview-tiles {
    margin-bottom: 0;
    /* display: inline-block; */
    overflow-x: scroll;
    white-space:nowrap;
  }

  /* .preview-tile {
    display:inline-block;

  } */

  .preview-control-icons {
    width: 40%;
  }
}

@media screen and (min-width: 1000px) {

  .preview-controls-container {
    margin-right: -400px;
  }

  .preview-svg{
    margin-right: 5px;
  }

}

.preview-controls > .randomize.button {
  margin-right: 0;
  width: 120px;
}