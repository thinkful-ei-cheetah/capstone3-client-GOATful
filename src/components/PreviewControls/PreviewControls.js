import React from 'react';
import PreviewItem from '../PreviewItem/PreviewItem'
import './PreviewControls.css'

export default function PreviewControls({ prevList, selected, previewClick , editClick, delClick}) {
  let renderPreviews = () => {
    return prevList.map(preview => {
      console.log(preview.id)
      console.log(selected.id)
      if (preview.id !== selected.id) { 
      
        return (
          <PreviewItem
            key={preview.id}
            preview={preview}
            previewClick={previewClick}
          />
        )
      } else {
      console.log("i am elsed")
        return (
          <PreviewItem
            key={preview.id}
            preview={preview}
            editClick={editClick}
            delClick={delClick}
          />
        )
      }
    })
  }

  let horizontalScroll = (e) => {
<<<<<<< HEAD
    // e.preventDefault();
    // window.scrollTo(0, 0);
=======
>>>>>>> 9003bed0344b0aaaae6c8ebd3b255ea6dbdd4ba3
    const left = document.getElementById('hScroll')
    left.scrollLeft += e.deltaY
    
    }

  let handleMouseIn = (e) => {
    document.body.style.overflow = 'hidden';
  }

  let handleMouseOut = (e) => {
    document.body.style.overflow = 'auto';
  }

  return (
    <div className="preview-controls-container">
      <h2 className="preview-title">Previews</h2>
      <div 
      className="preview-tiles" 
      id="hScroll" 
      onWheel={(e) => horizontalScroll(e)}
      onMouseEnter={(e)=> handleMouseIn(e)}
      onMouseLeave={(e)=> handleMouseOut(e)}
      >
        
        {(!selected) ? false : renderPreviews()}
      </div>
<<<<<<< HEAD
    </div>
=======
>>>>>>> 9003bed0344b0aaaae6c8ebd3b255ea6dbdd4ba3
  )
}

PreviewControls.defaultProps = {
  prevList: [],
  selected: {},
  previewClick: () => {},
  editClick: () => {},
  deleteClick: () => {}
}