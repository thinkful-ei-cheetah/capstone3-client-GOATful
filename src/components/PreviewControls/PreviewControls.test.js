import React from 'react';
import ReactDOM from 'react-dom';
import PreviewControls from './PreviewControls';
import renderer from 'react-test-renderer';


describe('PreviewControls Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(< PreviewControls />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<PreviewControls/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
})
