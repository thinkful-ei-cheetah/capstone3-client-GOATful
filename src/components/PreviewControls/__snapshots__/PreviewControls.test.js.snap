// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreviewControls Component renders the UI as expected 1`] = `
<div
  className="preview-controls-container"
>
  <h2
    className="preview-title"
  >
    Previews
  </h2>
  <div
    className="preview-tiles"
    id="hScroll"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onWheel={[Function]}
  />
</div>
`;
